### PR TITLE
Fixed broken links (#7336)

### PR DESCRIPTION
**Pull request by hdhalter**
* fixed broken links

Signed-off-by: Heather Halter <hdhalter@amazon.com>

* Update index.md

Removed py-ml client from list since it doesn't exist i 2.3

Signed-off-by: Heather Halter <HDHALTER@AMAZON.COM>

* Update viz-index.md

Fixed a typo

Signed-off-by: Heather Halter <HDHALTER@AMAZON.COM>

* Update docker.md

Fixed typo

Signed-off-by: Heather Halter <HDHALTER@AMAZON.COM>

---------

Signed-off-by: Heather Halter <hdhalter@amazon.com>
Signed-off-by: Heather Halter <HDHALTER@AMAZON.COM>
Co-authored-by: Heather Halter <hdhalter@amazon.com>
(cherry picked from commit fa00dbec534feab5c88ec55e7fd8d657ab000c5f)

### Description
_Describe what this change achieves._

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._

### Version
_List the OpenSearch version to which this PR applies, e.g. 2.14, 2.12--2.14, or all._

### Checklist
- [ ] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by kolchfa-aws — APPROVED** [review]
LGTM

**Comment by peterzhuamazon** [comment]
After comment out the redirect urls, there are 10 more dead links to fix:
```
`verify': Found 10 dead links: (RuntimeError)
https://opensearch.org/docs/latest/dashboards/maps-plugin/, linked to in ./_dashboards/visualize/viz-index.md
https://opensearch.org/docs/latest/search-plugins/knn/inswz/, linked to in ./_api-reference/script-apis/create-stored-script.md
https://opensearch.org/docs/latest/opensearch/query-dsl/full-text#multi-match, linked to in ./_opensearch/ux.md
https://opensearch.org/docs/latest/tools/index/, linked to in ./_about/quickstart.md
https://opensearch.org/docs/latest/tools/logstash/index/, linked to in ./_clients/logstash/index.md, ./_clients/logstash/read-from-opensearch.md
https://opensearch.org/docs/latest/tools/logstash/index#install-logstash, linked to in ./_clients/logstash/ship-to-opensearch.md
https://opensearch.org/docs/latest/dashboards/get-started/quickstart-dashboards/, linked to in ./_install-and-configure/upgrade-opensearch/appendix/rolling-upgrade-lab.md, ./_dashboards/discover/index-discover.md, ./_observing-your-data/log-ingestion.md, ./_about/quickstart.md
https://opensearch.org/docs/latest/opensearch/snapshots/snapshot-restore/#security-plugin-considerations, linked to in ./_tuning-your-cluster/availability-and-recovery/snapshots/sm-api.md
https://opensearch.org/docs/latest/tools/index/#compatibility-matrices, linked to in ./_install-and-configure/upgrade-opensearch/index.md
https://opensearch.org/docs/latest/opensearch/query-dsl/full-text#advanced-filter-options, linked to in ./_opensearch/ux.md
```

**Comment by peterzhuamazon** [comment]
Just to add context why those two redirect are commented out:
```

LinkChecker: [Info] Checking /clients/java-rest-high-level
                    ------------------------------------------------
      Jekyll 4.3.3   Please append `--trace` to the `build` command
                     for any additional information or backtrace.
                    ------------------------------------------------
/opt/homebrew/lib/ruby/gems/3.2.0/gems/jekyll-4.3.3/lib/jekyll/stevenson.rb:7:in `block in initialize': stack level too deep (SystemStackError)
	from /opt/homebrew/Cellar/ruby@3.2/3.2.4/lib/ruby/3.2.0/logger.rb:743:in `format_message'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/jekyll-4.3.3/lib/jekyll/stevenson.rb:28:in `add'
	from /opt/homebrew/Cellar/ruby@3.2/3.2.4/lib/ruby/3.2.0/logger.rb:695:in `info'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/jekyll-4.3.3/lib/jekyll/log_adapter.rb:148:in `public_send'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/jekyll-4.3.3/lib/jekyll/log_adapter.rb:148:in `write'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/jekyll-4.3.3/lib/jekyll/log_adapter.rb:66:in `info'
	from /Users/zhujiaxi/amazonworks/sourcecodes/opensearch-project/documentation-website/_plugins/link-checker.rb:249:in `check_internal'
	from /Users/zhujiaxi/amazonworks/sourcecodes/opensearch-project/documentation-website/_plugins/link-checker.rb:228:in `check'
	 ... 7485 levels...
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/bundler-2.2.23/lib/bundler/friendly_errors.rb:128:in `with_friendly_errors'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/bundler-2.2.23/exe/bundle:37:in `<top (required)>'
	from /opt/homebrew/opt/ruby@3.2/bin/bundle:25:in `load'
	from /opt/homebrew/opt/ruby@3.2/bin/bundle:25:in `<main>'
```

File: _clients/java-rest-high-level.md [hunk 1]
```diff
@@ -3,7 +3,7 @@ layout: default
 title: Java high-level REST client
 nav_order: 60
 nav_exclude: true
-redirect_to: https://opensearch.org/docs/latest/clients/java-rest-high-level/
+#redirect_to: https://opensearch.org/docs/latest/clients/java-rest-high-level/
```
Comment (kolchfa-aws) [inline _clients/java-rest-high-level.md:6]:
```suggestion
redirect_to: https://opensearch.org/docs/latest/clients/java-rest-high-level/
```

File: _clients/java.md [hunk 2]
```diff
@@ -3,7 +3,7 @@ layout: default
 title: Java client
 nav_order: 65
 nav_exclude: true
-redirect_to: https://opensearch.org/docs/latest/clients/java/
+#redirect_to: https://opensearch.org/docs/latest/clients/java/
```
Comment (kolchfa-aws) [inline _clients/java.md:6]:
```suggestion
redirect_to: https://opensearch.org/docs/latest/clients/java/
```

**Comment by dblock** [comment]
The link checker is failing because in `_about/breaking-changes.md` there's a link to `java-rest-high-level.md` which has a `redirect_to: https://opensearch.org/docs/latest/clients/java-rest-high-level/`, which is a redirect to itself if you deploy this code to `latest`. The reason why the other such redirects aren't a problem is because nothing links to these pages, so the links don't get checked.

The quickest workaround is to remove the link in `_about/breaking-changes.md`.

I found which page was linking to this by making a change in the link checker log as well. You should probably make it in your main for the next problem.

```
--- a/_plugins/link-checker.rb
+++ b/_plugins/link-checker.rb
@@ -246,7 +246,7 @@ module Jekyll::LinkChecker
   # Check if an internal link is accessible
 
   def self.check_internal(url, location)
-    Jekyll.logger.info "LinkChecker: [Info] Checking #{url}".cyan
+    Jekyll.logger.info "LinkChecker: [Info] Checking #{url} (#{location})".cyan
     return true if @ignored_paths =~ url
```

**Pull request merged**